[MRG + 1] Centralize one-class SVM docs

TheAtomicOption opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs

Example: Fixes #10923. Includes #10967, a previous less comprehensive PR.

#### What does this implement/fix? Explain your changes.

Reduces the description of One-Class SVM class and usage in the SVM docs to a one sentence description, and instead references the more comprehensive documentation on novelty and outliers.

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

TheAtomicOption committed:
Condensed OneClassSVM description

Implements issue #10923
Retained reference to the class, but condensed the description to a single sentence. 
Set reference to the Novelty Detection section in outliers that references.

TheAtomicOption committed:
Added example

Added the example that was removed from the SVM description page as part of #10923

TheAtomicOption committed:
doc string points to comprehensive coverage

Updated User Guide link to point to where the topic is comprehensively covered.

TheAtomicOption commented:
Ok. Most recent (3rd) commit in this PR should hopefully do that. 

I'm still a bit fuzzy on how reStructuredText converts :ref:'some_title' to the correct link. Hopefully I mimic'd the syntax well enough and it came out correct.

amueller commented on sklearn/svm/classes.py:
```diff
@@ -1009,7 +1009,7 @@ class OneClassSVM(BaseLibSVM, OutlierMixin):
     The implementation is based on libsvm.
 
-    Read more in the :ref:`User Guide <svm_outlier_detection>`.
+    Read more in the :ref:`User Guide <novelty_detection>`.
```
I think you have to add an anchor in the place you want to reference
```
.. _novelty_detection:
```
above the header...

TheAtomicOption committed:
fix hyperlink reference in svm doc

implicit section heading didn't link, so using explicit instead

TheAtomicOption committed:
fix hyperlink in classes doc

implicit hyperlink target didn't work so using explicit target

amueller commented on sklearn/svm/classes.py:
```diff
@@ -1009,7 +1009,7 @@ class OneClassSVM(BaseLibSVM, OutlierMixin):
     The implementation is based on libsvm.
 
-    Read more in the :ref:`User Guide <svm_outlier_detection>`.
+    Read more in the :ref:`User Guide <outlier_detection>`.
```
that also works.

amueller reviewed (approved):
LGTM

TheAtomicOption committed:
removed unneeded word

explicit reference pulls in section name which doesn't require the preceding word 'section'

qinhanmin2014 reviewed (approved):
LGTM

glemaitre committed:
nitpicks

glemaitre commented:
Merging. I just jump a line to have less than 80 characters.

@TheAtomicOption thanks!!!
